### DISCUSSION TITLE
Oracle InstantClient in Docker

### DISCUSSION THREAD

**Discussion by Jika97** [apache/superset#22980] [General]
Good morning,

I'm not very familiar with Docker but I installed Apache Superset on a server because it's easy to deploy it quickly.

My data source comes from an Oracle database, I understood that it was necessary to install the instantclient to be able to access my database, something that I manage to do by entering directly into the /app container .

Sometimes I shut down and restart my containers. I would like to know where **exactly** should I place in the DockerFile this command: 

`pip install cx_oracle && apt-get install alien libaio1 -y && wget https://download.oracle.com/otn_software/linux/instantclient/218000 /oracle-instantclient-basic-21.8.0.0.0-1.x86_64.rpm && alien --to-deb oracle-instantclient-basic-21.8.0.0.0-1.x86_64.rpm && dpkg -i oracle-instantclient- basic_21.8.0.0.0-2_amd64.deb && rm -rf oracle*`

So that at each restart it can take into account the instantclient.

I am convinced that this aid will help more than one person.

**Comment by paulobitar** [comment]
> Good morning,
> 
> I'm not very familiar with Docker but I installed Apache Superset on a server because it's easy to deploy it quickly.
> 
> My data source comes from an Oracle database, I understood that it was necessary to install the instantclient to be able to access my database, something that I manage to do by entering directly into the /app container .
> 
> Sometimes I shut down and restart my containers. I would like to know where **exactly** should I place in the DockerFile this command:
> 
> `pip install cx_oracle && apt-get install alien libaio1 -y && wget https://download.oracle.com/otn_software/linux/instantclient/218000 /oracle-instantclient-basic-21.8.0.0.0-1.x86_64.rpm && alien --to-deb oracle-instantclient-basic-21.8.0.0.0-1.x86_64.rpm && dpkg -i oracle-instantclient- basic_21.8.0.0.0-2_amd64.deb && rm -rf oracle*`
> 
> So that at each restart it can take into account the instantclient.
> 
> I am convinced that this aid will help more than one person.

In your requirements-local.txt , put: cx_Oracle==8.3.0

Under the superset folder, create a folder (I named it oracle_client)  and put the instant_client on it

If you don´t have a Dockerfile, create one  (under the superset folder) and write :

USER root

RUN apt-get update -y --fix-missing
RUN apt-get install -y unzip
RUN mkdir /opt/oracle
COPY oracle_client/instantclient-basiclite-linux.x64-21.5.0.0.0dbru.zip /opt/oracle/instantclient.zip 
ENV ORACLE_HOME /opt/oracle/instantclient_21_5
WORKDIR /opt/oracle
RUN unzip instantclient.zip 
RUN ln -s libclntsh.so.12.1 libclntsh.so
RUN ln -s libocci.so.12.1 libocci.so
RUN apt-get install libaio-dev libaio1
RUN sh -c "echo /opt/oracle/instantclient_21_5 > /etc/ld.so.conf.d/oracle-instantclient.conf"
RUN ldconfig
ENV LD_LIBRARY_PATH=/opt/oracle/instantclient_21_5:$LD_LIBRARY_PATH
ENV PATH=/opt/oracle/instantclient_21_5:$PATH

USER superset

I Hope it helps you
